### PR TITLE
Add build for Rancher Desktop embed of dashboard

### PR DESCRIPTION
**Pull request by rak-phillip**
This allows us to release dashboard artifacts that are meant to be embedded into Rancher Desktop. My current approach for achieving this goal is to trigger a build when a tag matches the given pattern `v**-desktop*` (e.g. v2.6.3-desktop). 

This build sets the variable `RANCHER_ENV=desktop`, creates archive of the build, generates a checksum, and uploads the artifacts to the release. Rancher Desktop will then be able to download these artifacts at build-time and bundle a customized version of Dashboard into the final product. 

Example release from my fork:

https://github.com/rak-phillip/dashboard/releases/tag/v2.6.3-desktop.11

supports rancher-sandbox/rancher-desktop#1192

### CODE REVIEW

**Comment by vincent99** [comment]
I'm not necessarily opposed to switching (nor.. deciding at this point...), but...  Why build desktop with github actions and publish to github releaess, and all other versions with drone to google bucket?

**Comment by rak-phillip** [comment]
I saw it as a quick way to proof things out and iron out any issues with this build without too much impact for the rest of the project. If this isn't desired, we can implement the same via drone.

File: .github/workflows/release-rancher-desktop.yml
Comment (richard-cox) [inline .github/workflows/release-rancher-desktop.yml:28]: I think, not sure, it's normally better practise to pull these sections out in to their own file. There's some example bash files in `./scripts` that drone uses. The file could live in `.github/workflows/scripts`

File: .github/workflows/release-rancher-desktop.yml
Comment (richard-cox) [inline .github/workflows/release-rancher-desktop.yml:6]: Going forward we may use the approach for other products so it may make sense to group them by product name first instead - `desktop-v*`. This would make it easier to dig through releases in the github ui and it would avoid lots of tags with non-dashboard versions first

**Review by richard-cox — CHANGES_REQUESTED** [review]
Two minor comments. I'd also echo Vince's point about having multiple build processes. We should decide which we should use and then stick to it going forward. My 2 pence... github actions and releases are fine.

Edit - Do we need to cater for pre-release/release settings for the github release?

**Review by nwmac — APPROVED** [review]
LGTM - agreed to switch version after desktop label

File: .github/workflows/release-rancher-desktop.yml
Comment (rak-phillip) [reply .github/workflows/release-rancher-desktop.yml]: Sounds like a good approach to me - fixed

File: .github/workflows/release-rancher-desktop.yml
Comment (rak-phillip) [reply .github/workflows/release-rancher-desktop.yml:28]: I've done it in other areas, for instance in [our build script for Steve](https://github.com/rancher-sandbox/steve/blob/main/bin/build.sh). I think that it would make sense to break out into separate build script if we added any more complexity.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**
